common/src/main/java/net/adoptopenjdk/icedteaweb/io/FileUtils.java

```diff
@@ -240,6 +240,8 @@ public static void deleteWithErrMesg(File f) {
     private static void createRestrictedFile(File file, boolean isDir) throws IOException {
 
         File tempFile = new File(file.getCanonicalPath() + ".temp");
+        FileUtils.deleteWithErrMesg(tempFile, "Could not delete [" + tempFile + "]");
```

Comment:
maybe it would be better to do this at the end of the method to prevent race conditions where two threads try to create the same file